repos/system_upgrade/el7toel8/actors/checkfirstpartitionoffset/libraries/check_first_partition_offset.py

```diff
@@ -0,0 +1,30 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import GRUBDevicePartitionLayout
+
+SAFE_OFFSET_BYTES = 1024*1024  # 1MiB
+
+
+def check_first_partition_offset():
+    problematic_devices = []
+    for grub_dev in api.consume(GRUBDevicePartitionLayout):
+        first_partition = min(grub_dev.partitions, key=lambda partition: partition.start_offset)
+        if first_partition.start_offset <= SAFE_OFFSET_BYTES:
+            problematic_devices.append(grub_dev.device)
+
+    if problematic_devices:
+        summary = ('GRUB devices with too little space reserved before the first partition were found. '
+                   'Such devices can present a problem when updating the boot loader. Problematic devices: \n'
+                   '{0}')
+        problematic_devices_fmt = ['- {0}'.format(dev) for dev in problematic_devices]
+
+        hint = 'Reformat the devices so that there is at least {0} kiB space before the first partition.'
```

Comment:
```suggestion
        summary = (                                                            
            'On the system booting by using BIOS, the in-place upgrade fails ' 
            'when upgrading the GRUB2 bootloader if the boot disk\'s embedding area '
            'does not contain enough space for the core image installation. '  
            'This results in a broken system, and can occur when the disk has been '
            'partitioned manually, for example using the RHEL 6 fdisk utility.\n\n'
                                                                  
            'The list of devices with small embedding area:\n'    
            '{0}.'    
        )    
        problematic_devices_fmt = ['- {0}'.format(dev) for dev in problematic_devices]
    
        hint = (    
            'We recommend to perform a fresh installation of the RHEL 8 system '
            'instead of performing the in-place upgrade.\n'    
            'Another possibility is to reformat the devices so that there is ' 
            'at least {0} kiB space before the first partition. '    
            'Note that this operation is not supported and does not have to be '
            'always possible.'    
        )    
```